src/libraries/System.Text.Json/gen/JsonSourceGenerator.Roslyn4.0.cs

```diff
@@ -30,42 +31,35 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     (node, _) => node is ClassDeclarationSyntax,
                     (context, _) => (ClassDeclarationSyntax)context.TargetNode);
 
-            IncrementalValueProvider<(Compilation, ImmutableArray<ClassDeclarationSyntax>)> compilationAndClasses =
-                context.CompilationProvider.Combine(classDeclarations.Collect());
+            IncrementalValueProvider<SourceGenerationSpec?> sourceGenSpec = context.CompilationProvider
+                .Combine(classDeclarations.Collect())
+                .Select(static (tuple, cancellationToken) =>
+                {
+                    Parser parser = new(tuple.Left);
+                    return parser.GetGenerationSpec(tuple.Right, cancellationToken);
```

Comment:
Nit: both for the parser and the emitter, would it be cleaner to just have an exposed static method?  Seems like the use is effectively `new Parser(...).GetGenerationSpec(...)`, with the caller not actually needing the instance reference other than for invoking the single method.

Reply:
Yes, I plan to do this on follow-up PRs (and also remove all the System.Reflection wrappers over the Symbols API) after this one is merged.